cli/core/utils.go

```diff
@@ -272,42 +272,67 @@ func WriteOutputToFileOrStdout(output []byte, out *string) {
 	}
 }
 
-func FilterNotCheckpointedOrWithdrawnValidators(
-	allValidatorsForEigenpod []ValidatorWithIndex,
-	onchainInfo []onchain.IEigenPodValidatorInfo,
+func SelectCheckpointableValidators(
+	client *ethclient.Client,
+	eigenpodAddress string,
+	validators []ValidatorWithIndex,
```

Comment:
hmm, is this better?

it's a bit simpler to have this function be stateless and literally just a filter, than to have it data-fetch as well

e.g if we end up calling two of these throughout a command by accident we're going to data fetch twice, and those RPC calls get rate-limited 

if you feel passionately about this sure but idk

Reply:
The things I like better about this:
1. The naming is a lot more clear
2. All the select/filter methods behave the same way

I hear you on this:

> e.g if we end up calling two of these throughout a command by accident we're going to data fetch twice, and those RPC calls get rate-limited

But I specifically looked at the places these were getting called and this behaves identically to the previous implementation. If we change things in the future we might want to change how this works, but for now - this felt fine